subsys/mgmt/mcumgr/Kconfig

```diff
@@ -1,19 +1,250 @@
+	help
+	  Enable verbose logging during a firmware upgrade.
+
+config IMG_MGMT_DUMMY_HDR
+	bool "Return dummy image header data for imgr functions"
+	help
+	  Returns dummy image header data for imgr functions,
+	  useful when there are no images present, Eg: unit tests.
+endif
+
+menuconfig MCUMGR_CMD_LOG_MGMT
+	bool "Enable mcumgr handlers for log management"
+	help
+	  Enables mcumgr handlers for log management
+
+if MCUMGR_CMD_LOG_MGMT
+config LOG_MGMT_CHUNK_LEN
+	int "Maximum chunk size for log downloads"
+	default 512
+	help
+	  Limits the maximum chunk size for log downloads, in bytes.  A buffer of
+	  this size gets allocated on the stack during handling of the log show
+	  command.
+
+config LOG_MGMT_NAME_LEN
+	int "Maximum log name length"
+	default 64
+	help
+	  Limits the maximum length of log names, in bytes.  If a log's name length
+	  exceeds this number, it gets truncated in management responses.  A buffer
+	  of this size gets allocated on the stack during handling of all log
+	  management commands.
+
+config LOG_MGMT_BODY_LEN
+	int "Maximum log body length"
+	default 128
+	help
+	  Limits the maximum length of log entry bodies, in bytes.  If a log
+	  entry's body length exceeds this number, it gets truncated in management
+	  responses. A buffer of this size gets allocated on the stack during
+	  handling of the log show command.
+
+config LOG_READ_WATERMARK_UPDATE
+	bool "Enable reading of log watermark update"
+	help
+	  Enables reading of log watermark update.
+
+endif
+
+menuconfig MCUMGR_CMD_OS_MGMT
+	bool "Enable mcumgr handlers for OS management"
+	select REBOOT
+	help
+	  Enables mcumgr handlers for OS management
+
+if MCUMGR_CMD_OS_MGMT
+config OS_MGMT_RESET_MS
+	int "Delay before executing reset command (ms)"
+	default 250
+	help
+	  When a reset command is received, the system waits this many milliseconds
+	  before performing the reset.  This delay allows time for the mcumgr
+	  response to be delivered.
+
+config OS_MGMT_TASKSTAT
+	bool "Support for taskstat command"
+	default y
+
+config OS_MGMT_ECHO
+	bool "Support for echo command"
+	default y
+endif
+
+
+menuconfig MCUMGR_CMD_STAT_MGMT
+	bool "Enable mcumgr handlers for statistics management"
+	depends on STATS
+	help
+	  Enables mcumgr handlers for statistics management.
+
+config STAT_MGMT_MAX_NAME_LEN
+	int "Maximum stat group name length"
+	default 32
+	depends on MCUMGR_CMD_STAT_MGMT
+	help
+	  Limits the maximum stat group name length in mcumgr requests, in bytes.
+	  A buffer of this size gets allocated on the stack during handling of all
+	  stat read commands.  If a stat group's name exceeds this limit, it will
+	  be impossible to retrieve its values with a stat show command.
+
+endmenu
 
 config MCUMGR_SMP_BT
 	bool "Bluetooth mcumgr SMP transport"
-	select MCUMGR
 	select BT
 	select BT_PERIPHERAL
 	select BT_GATT_DYNAMIC_DB
 	help
 	  Enables handling of SMP commands received over Bluetooth.
 
+if MCUMGR_SMP_BT
```

Comment:
This is reduntant with `depends on MCUMGR_SMP_BT` in `MCUMGR_SMP_BT_AUTHEN`.

Reply:
Done, thanks!